### ISSUE TITLE
DataGrid NRE on cell edit

### ISSUE THREAD

**Issue by aguahombre** [AvaloniaUI/Avalonia#2863]
Using build 0.8.999-cibuild0003548-beta.
I get an NRE in TextBox.cs when clicking to edit a DataGrid cell: 

![image](https://user-images.githubusercontent.com/17591364/63263672-28a54400-c281-11e9-96ad-195ac549a0b1.png)
The ```_presenter``` is null.

Here is my example code:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication10.MainWindow"
        Title="AvaloniaApplication10">
  <DataGrid Items="{Binding MyItems}">
    <DataGrid.Columns>
      <DataGridTextColumn Header="A" Binding="{Binding A}" Width="*" />
      <DataGridTextColumn Header="B" Binding="{Binding B}" Width="*" />
    </DataGrid.Columns>
  </DataGrid>
</Window>
```
```
using System.Collections.Generic;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace AvaloniaApplication10
{
    public class MainWindow : Window
    {
        public List<MyItem> MyItems { get; } = new List<MyItem>
        {
            new MyItem {A="Item A1", B="Item B1"},
            new MyItem {A="Item A2", B="Item B2"},
        };

        public MainWindow()
        {
            DataContext = this;
            AvaloniaXamlLoader.Load(this);
        }
    }

    public class MyItem
    {
        public string A { get; set; }
        public string B { get; set; }
    }
}
```

Click on any of the cells in the DataGrid to cause the NRE.
![image](https://user-images.githubusercontent.com/17591364/63263912-b7b25c00-c281-11e9-8e21-42ea7267034a.png)

**Comment by aguahombre** [comment]
Same code under Avalonia 0.8.1 gives
![image](https://user-images.githubusercontent.com/17591364/63264845-03660500-c284-11e9-96ad-8319cf75dec8.png)

**Comment by grokys** [comment]
I've just tried this here on master (by replacing `ControlCatalog.MainWindow` with your sample code) and I'm not seeing the error. I then tried creating a new avalonia app using the nightlies (`0.8.999-cibuild0004187-beta`) and I'm not seeing it there either.

Has this been fixed for you? Are you still seeing it? If so could you upload your project somewhere?

**Comment by aguahombre** [comment]
I have a repo that shows the bug in 0.8.999-cibuild0003548-beta.
The bug appears to be fixed in 0.8.999-cibuild0004187-beta.

**Comment by aguahombre** [comment]
Tested this again and works correctly in 0.9.10 so confirmed fixed
